src/global-environment.lisp

```diff
@@ -82,9 +87,15 @@
 (defun (setf var-derived-type) (new-value var)
   (check-type new-value ty)
   (let ((info (var-info var)))
-    (when (entry-derived-type info)
-      (style-warn "Overwriting derived type of ~S" var))
+    (let ((existing-derived-type (entry-derived-type info)))
+      (when (and existing-derived-type
+                 (not (type= existing-derived-type new-value)))
+        (style-warn "Overwriting derived type of ~S from ~A to ~A"
+                    var
+                    (unparse-type existing-derived-type)
+                    (unparse-type new-value))))
     (alexandria:when-let ((declared (var-declared-type var)))
+      (type= new-value declared) ;; TODO: use type= or remove
```

Comment:
what is this doing here? This WHEN is checking the declared type ^

Reply:
My bad. Forgot to remove debug statement :stuck_out_tongue_closed_eyes:

Reply:
Removed in be003aa